lib/ansible/modules/database/postgresql/postgresql_facts.py

```diff
@@ -0,0 +1,916 @@
+                repl_info["app_name"] = i[2]
+            else:
+                repl_info["app_name"] = ""
+
+            repl_info["client_addr"] = i[3]
+            if i[4]:
+                repl_info["client_hostname"] = i[4]
+            else:
+                repl_info["client_hostname"] = ""
+
+            repl_info["backend_start"] = i[5]
+            repl_info["state"] = i[6]
+
+            repl_dict[repl_pid] = repl_info
+
+        self.pg_facts["replications"] = repl_dict
+
+    def get_lang_info(self):
+        """
+        Get information about current supported languages.
+        """
+        query = ("SELECT l.lanname, a.rolname, l.lanacl "
+                 "FROM pg_language AS l "
+                 "JOIN pg_authid AS a ON l.lanowner = a.oid")
+        res = self.__exec_sql(query)
+        lang_dict = {}
+        for i in res:
+            lang_name = i[0]
+            lang_info = {}
+            lang_info["lanowner"] = i[1]
+            if i[2]:
+                lang_info["lanacl"] = i[2]
+            else:
+                lang_info["lanacl"] = ""
+
+            lang_dict[lang_name] = lang_info
+
+        self.pg_facts["languages"] = lang_dict
+
+    def get_namespaces(self):
+        """
+        Get information about namespaces.
+        """
+        query = ("SELECT n.nspname, a.rolname, n.nspacl "
+                 "FROM pg_catalog.pg_namespace AS n "
+                 "JOIN pg_authid AS a ON a.oid = n.nspowner")
+        res = self.__exec_sql(query)
+        nsp_dict = {}
+        for i in res:
+            nsp_name = i[0]
+            nsp_info = {}
+            nsp_info["nspowner"] = i[1]
+            if i[2]:
+                nsp_info["nspacl"] = i[2]
+            else:
+                nsp_info["nspacl"] = ""
+
+            nsp_dict[nsp_name] = nsp_info
+
+        self.pg_facts["namespaces"] = nsp_dict
+
+    def get_pg_version(self):
+        query = "SELECT version()"
+        raw = self.__exec_sql(query)[0][0]
+        self.pg_facts["version"] = raw.split()[1]
+
+    def get_db_info(self):
+        # Following query returns:
+        # Name, Owner, Encoding, Collate, Ctype, Access Priv, Size
+        query = ("SELECT d.datname, "
+                 "pg_catalog.pg_get_userbyid(d.datdba), "
+                 "pg_catalog.pg_encoding_to_char(d.encoding), "
+                 "d.datcollate, "
+                 "d.datctype, "
+                 "pg_catalog.array_to_string(d.datacl, E'\n'), "
+                 "CASE WHEN pg_catalog.has_database_privilege(d.datname, 'CONNECT') "
+                 "THEN pg_catalog.pg_size_pretty(pg_catalog.pg_database_size(d.datname)) "
+                 "ELSE 'No Access' END, "
+                 "t.spcname "
+                 "FROM pg_catalog.pg_database AS d "
+                 "JOIN pg_catalog.pg_tablespace t on d.dattablespace = t.oid")
+        res = self.__exec_sql(query)
+        db_dict = {}
+        for i in res:
+            db_name = i[0]
+            db_info = {}
```

Comment:
You can replace this whole block using a single assignment:
```python
                db_dict[db_name] = dict(
                    owner=i[1],
                    encoding=i[2],
                    collate=i[3],
                    ctype=i[4],
                    access_priv=i[5] if i[5] else '',
                    size=i[6],
                )
```

Reply:
this is the best hint I've recently viewed ;)